utils/create_dataset.py

```diff
@@ -138,3 +139,19 @@ def __getitem__(self, index):
                                 f"\nIgnore if some augmentations have padded with dontcare value.")
         sample['index'] = index
         return sample
+
+
+def define_raster_dataset(raster):
+    class RDataset(RasterDataset):
```

Comment:
It's great that we can use these out of the box. Ideally, we'd have a torchgeo dataset that accepts a pre-opened rasterio dataset though. Do you have an idea what effort would be needed to subclass RasterDataset and get it to work with a rasterio Dataset (ex. our VRTs) rather than a file on disk? I guess we have to rewrite the entire __init__ and __getitem__, right?

Reply:
at first sight, we need rewrite only __init__() and add one if statement when the raster is being read (if raster is a_file_path then resterio.open(raster) else move on).

Reply:
done, the VRTDataset is a subclass of the GeoDataset from TorchGeo